spec/controllers/api/v1x0/application_controller_spec.rb

```diff
@@ -64,4 +64,37 @@
       expect(response.status).to eq(401)
     end
   end
+
+  context "with entitlement enforcement" do
+    around do |example|
+      bypass_rbac do
+        example.call
+      end
+    end
+
+    let(:false_hash) do
+      false_hash = default_user_hash
+      false_hash["entitlements"]["hybrid_cloud"]["is_entitled"] = false
+      false_hash
+    end
+    let(:missing_hash) do
+      missing_hash = default_user_hash
+      missing_hash.delete("entitlements")
+      missing_hash
+    end
+
+    it "fails if the hybrid_cloud entitlement is false" do
+      headers =  { 'x-rh-identity' => encoded_user_hash(false_hash), 'x-rh-insights-request-id' => 'gobbledygook' }
+      get "#{api("1.0")}/portfolios", :headers => headers
+
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+    it "allows the request through if entitlements isn't present" do
+      headers = { 'x-rh-identity' => encoded_user_hash(missing_hash), 'x-rh-insights-request-id' => 'gobbledygook' }
+      get "#{api("1.0")}/portfolios", :headers => headers
```

Comment:
@lindgrenj6 SHouldn't this file be modified so that 
https://github.com/ManageIQ/manageiq-api-common/blob/810d63e3d983d765e064438bf55c942ff237acf7/spec/support/user_header_spec_helper.rb#L16

So that we can just use the default_header every where. For the one test where we want to check if its disabled we can dup that has and change the entitlement to false

Reply:
@mkanoor yeah that could work - that way we'd only be testing it in one place, and that way if we ever remove the functionality (or say that the `entitlements` are required) we could only change the test there.

@syncrou could we add this as a test to https://github.com/ManageIQ/manageiq-api-common/pull/59 ?

Reply:
@lindgrenj6 @mkanoor I can follow up on this in a new PR.  This code looks good.